tb_plugin/fe/src/utils/top.ts

```diff
@@ -30,8 +30,13 @@ export function useTopN(options?: IOptions) {
     ? React.useCallback(debounce(setActualTop, options.wait ?? 500), [])
     : setActualTop
   React.useEffect(() => {
-    setActualDebounce(useTop === UseTop.Use && top > 0 ? top : undefined)
-  }, [top, useTop])
+    setActualDebounce(useTop === UseTop.Use && topIsValid(topText) ? Number(topText): actualTop)
+  }, [topText, useTop])
 
-  return [top, actualTop, useTop, setTop, setUseTop] as const
+  return [topText, actualTop, useTop, setTopText, setUseTop] as const
+}
+
+export function topIsValid(topText: string) {
+  const top = Number(topText)
+  return top > 0 && Number.isInteger(top)
```

Comment:
need we us isNaN for top?

Reply:
For style I probably should. Technically NaN > 0 is false